misc/review/MidiStreamMessageBuilder.idl

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+// ============================================================================
+// This is part of the Windows MIDI Services App API and should be used
+// in your Windows application via an official binary distribution.
+// Further information: https://github.com/microsoft/MIDI/
+// ============================================================================
+
+#include "namespace_defines.h"
+#include "midl_defines.h"
+MIDI_IDL_IMPORT
+
+import "IMidiUniversalPacket.idl";
+import "MidiMessage128.idl";
+
+import "MidiFunctionBlockUIHintEnum.idl";
+import "MidiFunctionBlockMidi10Enum.idl";
+import "MidiFunctionBlockDirectionEnum.idl";
+import "MidiEndpointDiscoveryRequestsEnum.idl";
+import "MidiFunctionBlockDiscoveryRequestsEnum.idl";
+
+namespace MIDI_MIDL_NAMESPACE
+{
+    MIDI_API_CONTRACT_v1
+    static runtimeclass MidiStreamMessageBuilder
+    {
+        static IMidiUniversalPacket BuildEndpointDiscoveryMessage(
+            MIDI_TIMESTAMP timestamp,
+            UInt8 umpVersionMajor,
+            UInt8 umpVersionMinor,
+            MidiEndpointDiscoveryRequests requestFlags
+            );
+
+        static IMidiUniversalPacket BuildEndpointInfoNotificationMessage(
+            MIDI_TIMESTAMP timestamp,
+            UInt8 umpVersionMajor,
+            UInt8 umpVersionMinor,
+            Boolean hasStaticFunctionBlocks,
+            UInt8 numberOfFunctionBlocks,
+            Boolean supportsMidi20Protocol,
+            Boolean supportsMidi10Protocol,
```

Comment:
Weird that 2.0 comes before 1.0.

Reply:
The parameters are in order of spec fields. Probably not entirely necessary, but makes sense.

![image](https://github.com/microsoft/MIDI/assets/1421146/9855c5ad-5bb2-4924-b987-327bfb011e17)